### PR TITLE
WEBSITE-352 Only showing current bloggers in the side bar;

### PR DESCRIPTION
**Pull request by gunnarmorling**
Adding separate page with all bloggers

### CODE REVIEW

**Comment by emmanuelbernard** [comment]
`rake clean preview` crashes

```
An error during rendering /Users/emmanuel/Code/in.relation.to/Users/emmanuel/Code/in.relation.to/_partials/project/blog/post.html.haml occurred.
Please see /Users/emmanuel/Code/in.relation.to/_partials/project/blog/post.html for more information
An error during rendering /Users/emmanuel/Code/in.relation.to/Users/emmanuel/Code/in.relation.to/_partials/project/blog/post.html.haml occurred.
Please see /Users/emmanuel/Code/in.relation.to/_partials/project/blog/post.html for more information
An error during rendering /Users/emmanuel/Code/in.relation.to/Users/emmanuel/Code/in.relation.to/_partials/project/blog/post.html.haml occurred.
Please see /Users/emmanuel/Code/in.relation.to/_partials/project/blog/post.html for more information
An error during rendering /Users/emmanuel/Code/in.relation.to/Users/emmanuel/Code/in.relation.to/_partials/project/blog/post.html.haml occurred.
Please see /Users/emmanuel/Code/in.relation.to/_partials/project/blog/post.html for more information
An error during rendering /Users/emmanuel/Code/in.relation.to/Users/emmanuel/Code/in.relation.to/_partials/project/blog/post.html.haml occurred.
Please see /Users/emmanuel/Code/in.relation.to/_partials/project/blog/post.html for more information
An error during rendering /Users/emmanuel/Code/in.relation.to/Users/emmanuel/Code/in.relation.to/_partials/project/blog/blogger_list.html.haml occurred.
Please see /Users/emmanuel/Code/in.relation.to/_partials/project/blog/blogger_list.html for more information
```

**Comment by gunnarmorling** [comment]
Hum, indeed. Looking into it, but you can run `rake clean preview[development]` for now (that's how I used to test) to check the change.

**Comment by gunnarmorling** [comment]
@emmanuelbernard Rebased and force pushed. Editor mode is working again now with some minor simplications (e.g. no Gravatars next to posts etc.).

**Comment by hferentschik** [comment]
How are Gravatars related to the editor mode? I get that we in 'editor' mode we don't run the splitter for tags and authors, but is the information needed for Gravatars not coming via the author information provided via the data dir extension (btw, I realize that the splitter replaces/changes some existing mapped values, but I was not aware that the author information is affected as well. On this tangent, I would love to change the code around this whole splitting stuff. It is very hard to follow what's going on and changing the type of a field or mapped value causes problems in the template files as well).

**Comment by gunnarmorling** [comment]
Good catch. Tweaked that last commit once more. Gravatars are there now, only author/tag linking isn't functional as the splitters aren't working in editor mode. So to preview the author list, run in development mode.

Indeed the splitting stuff is a bit tough to grok, esp. the type changing. Not sure how exactly it could be improved; For me it's good enough for now, everything is working :)

**Comment by gunnarmorling** [comment]
Regarding ordering, in the side bar people with level 0 aren't shown, the others are ordered from level 1 to n (within one level alphabetically by name).

On the `/authors` page, all people are shown: First those with a level > 0 (again ordered by level and name), then all those with level 0 (ordered by name). I didn't see a reason to hide a blogger at all from the author list (we only imported those bloggers with at least one post).

**Comment by hferentschik** [comment]
> Indeed the splitting stuff is a bit tough to grok, esp. the type changing.

+1 I got caught multiple times by this

> Not sure how exactly it could be improved;

One would need to re-write this extensions and potentially how it integrates with Awestruct. I have a vague idea on how to do it, but it would take some time

> For me it's good enough for now, everything is working :)

Right. It would take too much time to change this now. Even though it is a bit of a maintenance time bomb. In a perfect world I would like to rewrite this extension and contribute it back to Awestruct. Or even better someone else would write a new/better splitter.

**Comment by hferentschik** [comment]
Out of interest, how did you determine the list of active bloggers? 
Regarding the `/authors` page - I am not a big fan of ordering by level. That is imo completely in-transparent for the viewer/user (unless we have some sort of indicator for the level). I think we should order alphabetically regardless on this page.

**Comment by hferentschik** [comment]
Apart of the ordering all looks fine. Just waiting with the merge to see whether someone else has an opinion regarding the order.

**Comment by gunnarmorling** [comment]
> Out of interest, how did you determine the list of active bloggers?

It's those where I felt they might be posting again (basically those related to Hibernate, plus Shane who also blogged within the last year). Anyone you miss or think should be removed? I don't mind changing it in one way or the other.

>  That is imo completely in-transparent for the viewer/user 

I don't really think that readers do care about the specific order very much. I think it makes sense to have the active ones first and then the others. Otherwise completely inactive bloggers show up first when solely going alphabetic.

**Comment by hferentschik** [comment]
> It's those where I felt they might be posting again (basically those related to Hibernate, plus Shane who also blogged within the last year

Ok

> Anyone you miss or think should be removed? 

I was just wondering

> I don't really think that readers do care about the specific order very much. I think it makes sense to have the active ones first and then the others

Personally I find it annoying if things are ordered in semi random manner. But I guess we can just merge for now.

**Comment by hferentschik** [comment]
Merged.
